examples/rust/extend_viewer_ui/src/main.rs

```diff
@@ -0,0 +1,198 @@
+
+#[tokio::main]
+async fn main() -> Result<(), Box<dyn std::error::Error>> {
+    // Direct calls using the `log` crate to stderr. Control with `RUST_LOG=debug` etc.
+    re_log::setup_native_logging();
+
+    // Install handlers for panics and crashes that prints to stderr and send
+    // them to Rerun analytics (if the `analytics` feature is on in `Cargo.toml`).
+    re_crash_handler::install_crash_handlers(re_viewer::build_info());
+
+    // Listen for TCP connections from Rerun's logging SDKs.
+    // There are other ways of "feeding" the viewer though - all you need is a `re_smart_channel::Receiver`.
+    let rx = re_sdk_comms::serve(
+        "0.0.0.0",
+        re_sdk_comms::DEFAULT_SERVER_PORT,
+        Default::default(),
+    )
+    .await?;
+
+    let native_options = eframe::NativeOptions {
+        app_id: Some("my_app_id".to_owned()),
+        ..re_viewer::native::eframe_options()
+    };
+
+    let startup_options = re_viewer::StartupOptions {
+        memory_limit: re_memory::MemoryLimit {
+            // Start pruning the data once we reach this much memory allocated
+            limit: Some(12_000_000_000),
+        },
+        persist_state: true,
+    };
+
+    // This is used for analytics, if the `analytics` feature is on in `Cargo.toml`
+    let app_env = re_viewer::AppEnvironment::Custom("My Wrapper".to_owned());
+
+    let window_title = "My Customized Viewer";
+    eframe::run_native(
+        window_title,
+        native_options,
+        Box::new(move |cc| {
+            let rx = re_viewer::wake_up_ui_thread_on_each_msg(rx, cc.egui_ctx.clone());
+
+            let re_ui = re_viewer::customize_eframe(cc);
+
+            let rerun_app = re_viewer::App::from_receiver(
+                re_viewer::build_info(),
+                &app_env,
+                startup_options,
+                re_ui,
+                cc.storage,
+                rx,
+            );
+
+            Box::new(MyApp { rerun_app })
+        }),
+    )?;
+
+    Ok(())
+}
+
+struct MyApp {
+    rerun_app: re_viewer::App,
+}
+
+impl eframe::App for MyApp {
+    fn save(&mut self, storage: &mut dyn eframe::Storage) {
+        // Store viewer state on disk
+        self.rerun_app.save(storage);
+    }
+
+    /// Called whenever we need repainting, which could be 60 Hz.
+    fn update(&mut self, ctx: &egui::Context, frame: &mut eframe::Frame) {
+        // First add our panel(s):
+        egui::SidePanel::right("my_side_panel")
+            .default_width(200.0)
+            .show(ctx, |ui| {
+                self.ui(ui);
+            });
+
+        // Now show the Rerun Viewer in the remaining space:
+        self.rerun_app.update(ctx, frame);
+    }
+}
+
+impl MyApp {
+    fn ui(&mut self, ui: &mut egui::Ui) {
+        ui.add_space(4.0);
+        ui.vertical_centered(|ui| {
+            ui.strong("My custom panel");
+        });
+        ui.separator();
+
+        if let Some(log_db) = self.rerun_app.log_db() {
```

Comment:
kinda out of scope, but if we'd expose the ViewerContext instead, we could do extra nice things like showing our data ui on on_hover 🤔